src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+to use a shortened version of the class name. In this example, we are going to
+use the class ``java.util.Scanner``, but we can refer to it as just
+``Scanner``. We could use the ``java.util.Scanner`` class without any problem
+and without any import statement, provided that we always referred to it by its
+full name.
+
+In Java, you can use any class that is available WITHOUT having to import the
+class, but you must adhere to two very important conditions:
+
+#. The ``javac`` and ``java`` commands must know that the class exists.
+#. You must use the full name of the class.
+
+.. admonition:: Try It
+
+   In the sample code, remove the ``import`` statement in line 1, and change
+   ``Scanner`` on lines 7 & 9 to ``java.util.Scanner``. The program should
+   still compile and run.
+
+The class naming system in Java is very hierarchical. The *full* name of the
+``Scanner`` class is really ``java.util.Scanner``. You can think of this name
+as having two parts:
+
+#. ``java.util`` is called the *package*,
+#. ``Scanner`` is the class name.
+
+We’ll talk more about the class naming system a bit later.
+
+Java Knows Which Classes Exist, Kinda
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+How do the ``java`` and ``javac`` commands know that certain classes
+exist? We have these rules:
+
+#. Java knows about all the classes you define in the ``.java`` and ``.class``
+   files in your current working directory.
+#. Java knows about all the classes that get shipped with the software.
+#. Java knows about all the classes included in your ``CLASSPATH`` environment
+   variable. Your ``CLASSPATH`` environment variable can name two kinds of
+   structures:
+
+   a. A *Java archive file*, ``.jar``, that contains Java classes.
+   b. Another Unix directory that contains Java class files.
+
+Declaring Variables
+--------------------
+
+In the example above, lines 5 - 7 contain variable declarations:
+
+.. sourcecode:: java
+   :lineno-start: 5
+
+   double fahrenheit;
+   double celsius;
+   Scanner input;
+
+Since Java is a statically typed language, we must always declare the data type
+for any variable. Lines 5 & 6 establish that ``fahrenheit`` and ``celsius``
+will hold values of type ``double``. In line 7, the variable ``input``
+references a ``Scanner`` object.
+
+If later in the code we try to initialize ``fahrenheit`` with a string:
+
+.. sourcecode:: java
+
+   fahrenheit = "xyz"
+
+the compiler throws an error because ``fahrenheit`` is supposed to be a double.
+
+The following error is common for new Java programmers. Suppose we forget to
+include the declaration for ``celsius``. What happens if we try to
+manually compile our program using ``javac TempConverter.java`` on the command
+line?
+
+.. admonition:: Try It
+
+   #. Edit your ``TempConverter`` class by removing line 6, which declares the
+      variable ``celsius``.
+   #. Using your terminal, navigate to the parent directory of your
```

Comment:
new update! since we've learned that a single file java program doesn't need to use the javac command to be filed now, I didn't include this instructions in their brief command line program. it's mentioned, but explicitly said that the command isn't necessary. we can demonstrate the same behavior here inside of intelliJ though with its compiler

Reply:
which is honestly more relevant since theyll be doing their debugging in the ide

Reply:
Updated.  Thanks.